libraries/AP_Param/AP_Param.cpp

```diff
@@ -3019,7 +3019,7 @@ bool AP_Param::add_param(uint8_t _key, uint8_t param_num, const char *pname, flo
 
     // check CRC
     auto &hinfo = const_cast<GroupInfo*>(info.group_info)[0];
-    const int32_t crc = *(const int32_t *)(&hinfo.def_value);
+    const auto crc = to_int32(hinfo.def_value);
```

Comment:
Why auto here? Feels like keeping it int32 is better for this case.

Reply:
Either or. Just following `almost always auto`. If you think `to_int32` is not clear enough to what the function does to what the resultant type is, then I'm happy to change. 

 https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/

Reply:
Changed to `int32_t` af264f5607f6d0e321574d9e20fb322e82bced71 here.